junction/proposals/management/commands/export_conf_proposals.py

```diff
@@ -0,0 +1,107 @@
+from django.core.management.base import BaseCommand
+from settings.common import ROOT_DIR
+
+from junction.base.constants import ProposalStatus
+from junction.conferences.models import Conference
+from junction.proposals.models import Proposal, ProposalSectionReviewerVoteValue
+from junction.proposals.permissions import is_conference_moderator
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--conference_slug",
+            default=None,
+            help="Enter the conference slug where to export reviewer votes from",
+        )
+
+        parser.add_argument("--user_id", default=None, help="Enter your user id")
+
+    def handle(self, *args, **options):
+
+        try:
+            conference = Conference.objects.get(slug=options.get("conference_slug"))
+            user = User.objects.get(id=options.get("user_id"))
+        except User.DoesNotExist:
+            self.stdout.write("Invalid user")
+            sys.exit(1)
+        except Conference.DoesNotExist:
+            self.stdout.write("Invalid conference")
+            sys.exit(1)
+
+        if not is_conference_moderator(user=user, conference=conference):
+            self.stdout.write("The user id is not a conference moderator")
+            sys.exit(1)
+
+        proposal_sections = conference.proposal_sections.all()
+        proposals_qs = Proposal.objects.select_related(
+            "proposal_type", "proposal_section", "conference", "author"
+        ).filter(conference=conference, status=ProposalStatus.PUBLIC)
+        proposals_qs = sorted(
+            proposals_qs, key=lambda x: x.get_reviewer_votes_sum(), reverse=True
+        )
+        proposal_vote_values = ProposalSectionReviewerVoteValue.objects.order_by(
+            "-vote_value"
+        )
+        vote_values_list = [v.vote_value for v in proposal_vote_values]
+        vote_values_desc = tuple(i.description for i in proposal_vote_values)
+        header = (
+            ("Proposal Type", "Title", "Sum of reviewer votes", "No. of reviewer votes")
+            + ("Public votes count", "Vote comments")
+            + tuple(vote_values_desc)
+        )
+
+        csv_contents = []
+
+        for section in proposal_sections:
+            section_proposals = [
+                p for p in proposals_qs if p.proposal_section == section
+            ]
+
+            for index, p in enumerate(section_proposals, 1):
+                vote_details = tuple(
+                    p.get_reviewer_votes_count_by_value(v) for v in vote_values_list
+                )
+                vote_comment = "\n".join(
+                    [
+                        comment.comment
+                        for comment in p.proposalcomment_set.filter(
+                            vote=True, deleted=False
+                        )
+                    ]
+                )
+                row = {
+                    header[0]: p.proposal_type.name,
```

Comment:
I think instead of `csv.DictWriter` you should simply use `csv.writer` and use lists to write each row. This reduces this complexity of creating a dictionary in each iteration and makes the code simpler. e.g.

https://github.com/pythonindia/junction/blob/e8cefd7910f807873a9f0d7650f6868ccffbeac0/junction/proposals/dashboard.py#L326

Reply:
Using dictwriter would be a lot more accurate, right? Always ensuring which values go for which fields? Probably, never creating confusion for future devs on which key belongs to which key?

Reply:
Please rename the `vote_details` variable to `reviewer_votes_count` in that case